From ae9d7f81cf4fa94566ed817b924ea500276991d0 Mon Sep 17 00:00:00 2001
From: Ajay Panicker <apanicke@google.com>
Date: Fri, 11 May 2018 12:03:07 -0700
Subject: [PATCH] DO NOT MERGE: Check number of attributes before writing to a buffer

Bug: 73824150
Test: Compile
Change-Id: I2a28a503cd74758e707d1e591b55c278d2299f45
(cherry picked from commit f6db54f071f6974e18b10bb0c2cfcf397cd4c980)
CVE-2018-9418
---

diff --git a/btif/src/btif_rc.c b/btif/src/btif_rc.c
index b55f001..cc65983 100644
--- a/btif/src/btif_rc.c
+++ b/btif/src/btif_rc.c
@@ -42,6 +42,7 @@
 #include "btif_util.h"
 #include "bt_common.h"
 #include "device/include/interop.h"
+#include "log/log.h"
 #include "uinput.h"
 #include "bdaddr.h"
 #include "osi/include/list.h"
@@ -4714,6 +4715,12 @@
     bdcpy(rc_addr.address, btif_rc_cb[0].rc_addr);
 
     app_settings.num_attr = p_rsp->num_val;
+
+    if (app_settings.num_attr > BTRC_MAX_APP_SETTINGS) {
+      android_errorWriteLog(0x534e4554, "73824150");
+      app_settings.num_attr = BTRC_MAX_APP_SETTINGS;
+    }
+
     for (xx = 0; xx < app_settings.num_attr; xx++)
     {
         app_settings.attr_ids[xx] = p_rsp->p_vals[xx].attr_id;
